assets/src/edit-story/components/library/panes/text/textSets/textSets.js

```diff
@@ -83,7 +82,34 @@ function TextSets() {
   );
 
   useEffect(() => {
-    getTextSets().then(setTextSets);
+    window.clearTimeout(loadingTimeoutRef.current);
+    isLoadingRef.current = false;
+
+    setFilteredTextSets(selectedCat ? textSets[selectedCat] : allTextSets);
+    setRenderedTextSets([]);
+  }, [selectedCat, textSets, allTextSets]);
+
+  useEffect(() => {
+    if (isLoadingRef.current) {
+      return;
+    }
+
+    if (renderedTextSets.length < filteredTextSets.length) {
+      isLoadingRef.current = true;
+
+      loadingTimeoutRef.current = window.setTimeout(() => {
+        isLoadingRef.current = false;
+        setRenderedTextSets([
+          ...renderedTextSets,
+          filteredTextSets[renderedTextSets.length],
+        ]);
+      }, 10);
+    }
+  }, [renderedTextSets, filteredTextSets]);
+
+  useEffect(() => {
+    window.clearTimeout(loadingTimeoutRef.current);
+    isLoadingRef.current = false;
```

Comment:
I'm not following what this is for?

Reply:
It's gone, your refactor cleaned up all the kruft 🎉